tech-specs/src/explore/design/pos.md

```diff
@@ -0,0 +1,258 @@
+  A validator is not being considered for block creation and cannot receive any new delegations.
+- *pending*:
+  A validator has requested to become a *candidate*.
+- *candidate*:
+  A validator is considered for block creation and can receive delegations.
+
+For each validator (in any state), the system also tracks total bonded tokens as a sum of the tokens in their self-bonds and delegated bonds, less any unbonded tokens. The total bonded tokens determine their voting voting power by division by the `votes_per_token` [parameter](#system-parameters). The voting power is used for validator selection for block creation and is used in governance related activities.
+
+#### Validator actions
+
+- *become validator*:
+  Any account that is not a validator already and that doesn't have any delegations may request to become a validator. It is required to only provide a public consensus key. For the action applied in epoch `n`, the validator's state will be immediately set to *pending* and it will be set to *candidate* for epoch `n + pipeline_length`.
+- *deactivate*:
+  Only a *pending* or *candidate* validator account may *deactivate*. For this action applied in epoch `n`, the validator's account is set to become *inactive* in the epoch `n + pipeline_length`.
+- *reactivate*:
+  Only an *inactive* validator may *reactivate*. Similarly to *become validator* action, for this action applied in epoch `n`, the validator's state will be immediately set to *pending* and it will be set to *candidate* for epoch `n + pipeline_length`.
+- *self-bond*:
+  A validator may lock-up tokens into a [bond](#bonds) only for its own validator's address.
+- *unbond*:
+  Any self-bonded tokens may be partially or fully [unbonded](#unbond).
+- *withdraw unbond*:
+  Unbonded tokens may be withdrawn in or after the [unbond's epoch](#unbond).
+
+#### Active validator set
+
+From all the *candidate* validators, in each epoch the ones with the most voting power limited up to the `max_active_validators` [parameter](#system-parameters) are selected for the active validator set. The active validator set selected in epoch `n` is set for epoch `n + pipeline_length`.
+
+### Delegator
+
+A delegator may have any number number of delegations. Delegations are stored in [bonds](#bonds).
+
+#### Delegator actions
+
+- *delegate*:
+  An account which is not a validator may delegate tokens to any number of validators. This will lock-up tokens into a [bond](#bonds).
+- *undelegate*
+  Any delegated tokens may be partially or fully [unbonded](#unbond).
+- *withdraw unbond*:
+  Unbonded tokens may be withdrawn in or after the [unbond's epoch](#unbond).
+
+## Bonds
+
+A bond locks-up tokens from validators' self-bonding and delegators' delegations. For self-bonding, the delegator's address is equal to the validator's address and the source of the bond is the validator's account. Only validators can self-bond. For a bond created from a delegation, the delegator's account is the bond's source.
```

Comment:
this is kinda in conflict with earlier which says that delegators can't be validators

Reply:
It's the "the delegator's address is equal to the validator's address" part, right? It's meant to refer to the fields of a bond object, but I agree that's not clear. I'll rename the `delegator` field to `source` instead